app/scripts/directives/createSecret.js

```diff
@@ -143,6 +143,17 @@ angular.module("openshiftConsole")
         };
 
         var linkSecretToServiceAccount = function(secret) {
+          if (!$scope.serviceAccountsNames.contains($scope.newSecret.pickedServiceAccountToLink)) {
+            var alert = {
+              name: 'createAndLink',
+              data: {
+                type: "success",
+                message: "Secret " + secret.metadata.name + " was created but couldn't be linked with service account " + $scope.newSecret.pickedServiceAccountToLink + "."
```

Comment:
> Seems reasonable. These will be separate alert messages?

yes

> i'm not suggesting it be two separate UI actions the user can take, i'm just suggesting you treat it as two steps internally and report error w/ finer granularity.

yes I understand, but we dont have a good story how to model those steps so they will make sense. Not saying that we could not do it, just that we dont have time :'( But definitely worth for as a follow up ?!